src/expint.jl

```diff
@@ -0,0 +1,28 @@
+function expint(x::Float64)
+    if x >= 1.0
+        e = exp(-x/2)
+        xinv = inv(x)
+        a = evalpoly(xinv, (1.3319248966682697359e-30, 0.99999999999999993242, 37.702667724670064050, 542.50695317051747114, 3836.6700640600929730, 14346.114698914059563, 28502.272377756659505, 28835.636437354567322, 13473.368739266960527,2387.9894006598233823, 99.579680699439479278))
+        b = evalpoly(xinv, (1.0, 38.702667724669835916, 579.20962089531281339, 4344.4743494790396875, 17740.385816044920402, 40220.102972229127618, 49665.114336529855022, 31552.009191297117961, 9222.0904372933369941, 992.44893809992759951,22.340799371096865866))
+        return (a / b * e) * e
+    else
+        a = evalpoly(x, (1.0000000000000000001,0.16967259504587971303,0.027806394103770057485,0.0014502330283706966415,0.000067149786258297615728,-5.9741062104757485179e-8))
+        b = evalpoly(x, (1,0.41967259504587972849,0.077168987309683782070,0.0078440023532535002066,0.00044590727789287393878,0.000011509944564265827774))
+        p = x * a / b
+        return (-(Base.MathConstants.eulergamma + log(x)) + p)
+    end
+end
+
+function expint(x::Float32)
+    if x >= 1.0f0
+        e = exp(-x/2)
+        xinv = inv(x)
+        a = evalpoly(xinv, (2.0705072525187918062f-21,0.99999989577122096213f0,6.8826670492773880464f0,10.355328549799196519f0,2.6865809995992064807f0))
```

Comment:
Float32 don't need nearly this many digits.

Reply:
Also just a product of outputing minimax polynomial in double precision.